src/cart/crt_swim.c

```diff
@@ -363,6 +363,8 @@ static int crt_swim_send_message(struct swim_context *ctx, swim_id_t to,
 		D_GOTO(out, rc);
 	}
 out:
+	if (rc && rpc_req != NULL)
+		crt_req_decref(rpc_req);
```

Comment:
If you move decref here, please remove it from 345 line as in original patch.

Reply:
I'm going to remove this line, it is not one of the main fixes.

Reply:
just be clear, your patch includes many good fixes and cleanup, but also includes quite a few style cleanup. It's a little bit too risky to land it as one patch especially for 1.2. So my intention is really just taking out the most important fix and try to land it to 1.2 ASAP.

Reply:
Agree. :-( Sorry about that.